qemu/tests/qemu_guest_agent.py

```diff
@@ -4125,45 +4125,43 @@ def execute_qga_cmds_loop():
                 self.gagent.get_osinfo()
                 self.gagent.get_virtio_device()
 
-        def _get_handle(pid):
+        def _process_qga_resource():
             """
-            Check the handles of qga via handle.exe
+            Check the handles of process qemu-ga.
 
-            :params pid: process ID of qga.service
             :return qga_handles: the handles of qga.service
+            :return qga_memory: the memory of qga.service
             """
-            cmd_get_handle = params["cmd_get_qga_handle"] % pid
-            qga_handles = int(session.cmd_output(cmd_get_handle
-                                                 ).strip().split(":")[1])
-            return qga_handles
 
-        def _base_on_bg_check_handles():
+            get_resour = self.params['cmd_get_qga_resource']
+            qga_resources = session.cmd_output(get_resour).strip().split(" ")
+            qga_handles = qga_resources[0]
+            qga_memory = int(qga_resources[1]) / 1024
+            return (int(qga_handles), int(qga_memory))
+
+        def _base_on_bg_check_resource_leak():
             if bg.is_alive():
-                end_handles = _get_handle(pid_qga)
-                if end_handles > 300:
-                    test.fail("Handles is %s more than 300, qga-commands"
-                              " caused resource leak." % end_handles)
+                qga_handles, qga_memory = _process_qga_resource()
+                if qga_handles > 300 or qga_memory > 3000:
```

Comment:
Could you explain why > 300 or > 3000? thanks.

Reply:
I think we should consider handles/memory should maintain a stable value, but not only a limitation.

Reply:
I‘ve confirmed with Yan and googled it, but there is no official data about it. so I can just set it according to our actual situation.

Reply:
We don't need an official data.
We also need to consider the handle/memory should not be increased or increased with a little value.